tests/ctest/test_opts.c

```diff
@@ -1044,6 +1044,40 @@ static void test_help (void)
 
 	ksDel (ks);
 
+
```

Comment:
Please add a similar test case that commands (see further down in the file for an example) as well as deeply nested keys, so we know both features work together as well.

Reply:
Do you mean **sub**-commands combined with deeply nested keys? That is actually explicitly not supported.

See e.g. 

https://github.com/ElektraInitiative/libelektra/blob/5702ce0496823fdd12c4e1e6ff9d6d1ca1b71a40/tests/ctest/test_opts.c#L748-L762

and 

https://github.com/ElektraInitiative/libelektra/blob/5702ce0496823fdd12c4e1e6ff9d6d1ca1b71a40/tests/ctest/test_opts.c#L732-L746

which state: "in sub-command mode short/long options **without command metakey on direct parent should be illegal**" (emph. mine)

IMO if this should be changed, it is separate to issue #3999 which only deals with the generation of the help message for already supported specification hierarchies (sub-commands with deep nesting of options aren't supported).

Reply:
Ah okay, in that case your tests are fine.